src/profile/ProfileForm.tsx

```diff
@@ -290,96 +294,138 @@ export const ProfileFormConsumer: React.FC<ProfileConsumerFormProps> = (
-                  onClick={() => {
-                    props.context.setValue("password", nanoid(), true);
-                    // set the password visible so you can see what you get
-                    setShowPassword(true);
+                  onClick={async () => {
+                    const data = await generateToken({
+                      username: props.user!.username,
+                    });
+                    if (data.success) {
+                      setPasswordResetToken(data.data);
+                    } else {
+                      setGeneralErrors(data.error.detail);
+                    }
                   }}
                 >
-                  Generate
+                  Generate password reset token
                 </Button>
-              </Col>
-              <Col>
-                <Field
-                  errors={props.context.errors}
-                  type={showPassword ? "text" : "password"}
-                  name="password"
-                  ref={props.context.register({
-                    minLength: {
-                      value: 8,
-                      message: "USER.PASSWORD.TOO_SHORT.8",
-                    },
-                    required: newPasswordRequired
-                      ? "USER.PASSWORD.NEW_REQUIRED"
-                      : undefined,
-                    validate: (password?: string) => {
-                      // the pattern argument doesn't let us return an error if the value FAILS a regex,
-                      // so we'll do it ourselves
-                      if (password && /^\d*$/.test(password)) {
-                        return "USER.PASSWORD.ENTIRELY_NUMERIC";
-                      }
-                    },
-                  })}
-                  onChange={async () => {
-                    await props.context.triggerValidation("password");
-                  }}
-                  append={
-                    <Button
-                      variant={
-                        showPassword ? "outline-primary" : "outline-secondary"
-                      }
-                      onClick={() => {
-                        setShowPassword((show) => !show);
-                      }}
-                    >
-                      {showPassword ? (
-                        <FontAwesomeIcon icon={faEyeSlash} />
-                      ) : (
-                        <FontAwesomeIcon icon={faEye} />
-                      )}
-                    </Button>
-                  }
-                />
+                {generateTokenInfo.state === RequestState.Running && (
+                  <pre key="returned-token">Generating...</pre>
+                )}
+                {passwordResetToken && (
+                  <pre
+                    key="returned-token"
+                    className="user-select-all text-wrap text-break mt-1"
+                  >
+                    {resolve(
+                      config.baseurl,
+                      "reset_password",
```

Comment:
This appears to be missing the `help` portion of the URL, unless I'm misunderstanding how this is meant to be used.

Reply:
Ahh, you're right. fcf29c4